hack/test-go.sh

```diff
@@ -109,87 +111,107 @@ if [[ ${#testcases[@]} -eq 0 ]]; then
+    # coverage mode was used (set, count, or atomic). This line is included in
+    # each of the coverage profiles generated when running 'go test -cover', but
+    # we strip these lines out when combining so that there's only one.
+    echo "mode: ${KUBE_COVERMODE}"
+
+    # Include all coverage reach data in the combined profile, but exclude the
+    # 'mode' lines, as there should be only one.
+    for x in `find "${cover_report_dir}" -name "${cover_profile}"`; do
+      cat $x | grep -h -v "^mode:" || true
+    done
+  } >"${COMBINED_COVER_PROFILE}"
+
+  coverage_html_file="${cover_report_dir}/combined-coverage.html"
+  go tool cover -html="${COMBINED_COVER_PROFILE}" -o="${coverage_html_file}"
+  kube::log::status "Combined coverage report: ${coverage_html_file}"
+}
 
-# Create coverage report directories.
-cover_report_dir="/tmp/k8s_coverage/$(kube::util::sortable_date)"
-cover_profile="coverage.out"  # Name for each individual coverage profile
-kube::log::status "Saving coverage output in '${cover_report_dir}'"
-mkdir -p "${@+${@/#/${cover_report_dir}/}}"
-
-# Run all specified tests, collecting coverage results. Go currently doesn't
-# support collecting coverage across multiple packages at once, so we must issue
-# separate 'go test' commands for each package and then combine at the end.
-# To speed things up considerably, we can at least use xargs -P to run multiple
-# 'go test' commands at once.
-printf "%s\n" "${@}" | xargs -I{} -n1 -P${KUBE_COVERPROCS} \
-    go test "${goflags[@]:+${goflags[@]}}" \
-        ${KUBE_RACE} \
-        ${KUBE_TIMEOUT} \
-        -cover -covermode="${KUBE_COVERMODE}" \
-        -coverprofile="${cover_report_dir}/{}/${cover_profile}" \
-        "${cover_params[@]+${cover_params[@]}}" \
-        "${KUBE_GO_PACKAGE}/{}"
-
-combined_cover_profile="${cover_report_dir}/combined-coverage.out"
-{
-  # The combined coverage profile needs to start with a line indicating which
-  # coverage mode was used (set, count, or atomic). This line is included in
-  # each of the coverage profiles generated when running 'go test -cover', but
-  # we strip these lines out when combining so that there's only one.
-  echo "mode: ${KUBE_COVERMODE}"
-
-  # Include all coverage reach data in the combined profile, but exclude the
-  # 'mode' lines, as there should be only one.
-  for x in `find "${cover_report_dir}" -name "${cover_profile}"`; do
-    cat $x | grep -h -v "^mode:" || true
-  done
-} >"${combined_cover_profile}"
-
-coverage_html_file="${cover_report_dir}/combined-coverage.html"
-go tool cover -html="${combined_cover_profile}" -o="${coverage_html_file}"
-kube::log::status "Combined coverage report: ${coverage_html_file}"
-
-if [[ -x "${KUBE_GOVERALLS_BIN}" ]]; then
-  ${KUBE_GOVERALLS_BIN} -coverprofile="${combined_cover_profile}" || true
-fi
+runTestsForVersion() {
+  export KUBE_API_VERSION="$1"
+  runTests "${@:2}"
+}
+
+reportCoverageToCoveralls() {
```

Comment:
I'm not sure if factoring this out into its own subroutine was really necessary.